### ISSUE TITLE
[Feature] Optimize Logits Computation Speed via TP-Sharded Embedding

### ISSUE THREAD

**Issue by JamesBrianD** [sgl-project/sglang-jax#169]
### Checklist

- [x] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 2. Please use English, otherwise it will be closed.

### Motivation

The logits computation step in Transformers—projecting hidden states onto a large vocabulary space—can become a bottleneck due to the size of the embedding matrix (and its transpose, used in the final linear projection).

Sharding this embedding matrix via Tensor Parallelism (TP) would distribute both memory usage and computational load across TP ranks. This is particularly beneficial for models with large vocabularies, as it can speed up inference while improving scalability.

**Comment by JamesBrianD** [comment]
In addition to enabling tensor parallelism (TP) splitting embedding, this approach can also save TPU memory, which significantly increases the size and capacity of available KV cache.
